docs/src/_tutorials/logging.md

```diff
@@ -0,0 +1,284 @@
+layout: doc
+---
+
+## Logging
+
+Logging in meltano can be controlled via a standard yaml formatted [python logging dict config file](https://docs.python.org/3/library/logging.config.html#configuration-dictionary-schema).
+
+By default, meltano will look for this in a `logging.yaml` file in the project root. However, you can override this by
+setting the [environment variable](/guide/configuration#configuring-settings) `MELTANO_CLI_LOG_CONFIG` or by using the
+`meltano` CLI option `--log-config`. e.g. `meltano --log-config=my-prod-logging.yaml ...`.
+
+A logging.yaml contains a few key sections that you should be aware of.
+
+- `formatters` - This section contains the formatters that are used by the handlers. This controls the output format of the log messages (e.g. json).
+- `handlers` - This section contains the handlers which are used the loggers. This controls the output destination of the log messages (e.g. the console).
+- `loggers` - This section allows you to explicitly control specific module/class/etc named loggers.
+
+A few key points to note:
+
+1. Different handlers can use different formats.
+2. Different loggers can use different handlers and log at different log levels.
+3. We support all the [standard python logging handlers](https://docs.python.org/3/library/logging.handlers.html#) (e.g. rotating files, syslog, etc).
+
+Here's an annotated example of a logging.yaml file:
+
+```yaml
+version: 1
+disable_existing_loggers: false
+
+formatters:
+  default: # use a format similar to default generic python logging format
+    format: "[%(asctime)s] [%(process)d|%(threadName)10s|%(name)s] [%(levelname)s] %(message)s"
+  structured_plain: # log format for structured plain text logs without colored output
+    (): meltano.core.logging.console_log_formatter
+    colors: False
+  structured_colored: # log format for structured plain text logs WITH colored output
+    (): meltano.core.logging.console_log_formatter
+    colors: True
+  key_value: # log format for traditional key=value style logs
+    (): meltano.core.logging.key_value_formatter
+    sort_keys: False
+  json: # log format for json formatted logs
+    (): meltano.core.logging.json_formatter
+
+handlers:
+  console: # log to the console (stderr) using structured_colored formatter, logging everything at DEBUG level and up
+    class: logging.StreamHandler
+    level: DEBUG
+    formatter: structured_colored
+    stream: "ext://sys.stderr"
+  meltano_log: # log everything INFO and above to a file in the project root called meltano.log in json format
+    class: logging.FileHandler
+    level: INFO
+    filename: meltano.log
+    formatter: json
+  my_warn_file_handler: # log everything WARNING and above to automatically rotating log file in key_value format
+    class: logging.handlers.RotatingFileHandler
+    level: WARN
+    formatter: key_value
+    filename: /tmp/meltano_warn.log
+    maxBytes: 10485760
+    backupCount: 20
+    encoding: utf8
+
+root:
```

Comment:
Can you put in a note above (or as an inline comment) about `root`? In this example file is the first mention of it.

Reply:
Good catch. My brain immediately assumed "folks will know with the root logger is for" when that's definitely not the case.

Reply:
Added to description of all sections.